### DISCUSSION TITLE
increase indexing speed

### DISCUSSION THREAD

**Discussion by pycui** [milvus-io/milvus#36644] [Q&A and General discussion] [answered]
Right now using 8 index nodes with cpu: 16 and memory: 32Gi. When building index with HNSW m=64 and efConstruct=1024, it seems the speed is <30 million per hour. The CPU and memory utilization is not high, so wondering if there is other bottleneck, and generally wondering if there is way to speed up indexing.

Also wondering if there is plan to support multiple indexes on same field so that we don't have downtime when switching indexes.

@xiaofan-luan wondering if you can help triage this? thank you!

**Comment by xiaofan-luan** [comment]
I though you efConstruct is set to too large, usually I would start from 64 and see how the recall goes. This could dramatically increase the index build time. 

For index build, I thought It should easy to use all 16 core for each indexnodes. How much cpu does it take under your use case?

**Reply by xiaofan-luan** [reply]
We don't have a solid plan to support multiple index on one field but this is a great suggestion. Maybe it's a good timing to support it in milvus 3.0

**Reply by pycui** [reply]
It seems like the cpu util reported by attu was 15%-ish, but maybe it's still bottlenecked on cpu. I later found that increasing the indexNode replica correspondingly speed up the indexing, so this sort of solved our problem.

**Reply by pycui** [reply]
@xiaofan-luan thanks for the plan! Multi-index seems like will be very helpful. For now, what's the suggested workaround for switching indexes? Do we have to create new collection for this?

**Reply by xiaofan-luan** [reply, marked as answer]
Using a new collection, rebuild index on top of it and do a alias switch is good.
